### PR TITLE
Support Rails 4.x and fix generating examples

### PR DESCRIPTION
**Pull request by a2ikm**
I tweaked tests around `setup` and `teardown`, but not changed essentially.

I removed Gemfile.lock from repository because it can change for ActiveSupport's dependency on Test::Unit or minitest and CRuby's version. But it doesn't make any wrong effects on developing and releasing.

### CODE REVIEW

File: examples/applications/gemcutter/models/user.rb [hunk 1]
```diff
@@ -1,10 +1,20 @@
...
+      :through => :subscriptions,
+      :source  => :rubygem
+  else
+    has_many :rubygems, :through    => :ownerships,
+                        :order      => "name ASC",
+                        :conditions => { 'ownerships.approved' => true }
+    has_many :subscribed_gems, :through => :subscriptions,
+                               :source  => :rubygem,
+                               :order   => "name ASC"
+  end
```
Comment (kerrizor) [inline examples/applications/gemcutter/models/user.rb]: This is the sort of version switching I was dreading having to do... well done, thank you :)

File: examples/applications/spree/models/product.rb [hunk 2]
```diff
@@ -3,22 +3,41 @@ class Product < ActiveRecord::Base
...
+      :class_name => 'Variant',
+      :dependent => :destroy
+  else
+    has_many :variants,
+      :conditions => ["variants.is_master = ? AND variants.deleted_at IS NULL", false]
+    has_many :variants_including_master,
+      :class_name => 'Variant',
+      :conditions => ["variants.deleted_at IS NULL"],
+      :dependent => :destroy
+  end
```
Comment (kerrizor) [inline examples/applications/spree/models/product.rb]: Could we compress these three switched areas into one? I think it'd be easier to navigate in the future.

**Comment by kerrizor** [comment]
This is *fabulous* thank you! :heart:

**Comment by a2ikm** [comment]
@kerrizor I updated about switching in 6e3fd70. Should I squash it?

**Comment by kerrizor** [comment]
In this case, yes please :)

**Comment by a2ikm** [comment]
I squashed commits.
@kerrizor Could you check that?

**Comment by kerrizor** [comment]
Looks good from here! Merging and will cut a new release shortly.

Thanks again for the hard work here!

**Comment by a2ikm** [comment]
Thank you for merging!
I'm happy to use this gem with rails4!

**Pull request merged**